src/components/AccountSaas/SaasAccountContext.tsx

```diff
@@ -0,0 +1,12 @@
+import { createContext, ReactNode } from 'react'
+import { useSaasAccountProvider } from './useSaasAccountProvider'
+
+export const SaasAccountContext = createContext<ReturnType<typeof useSaasAccountProvider> | undefined>(undefined)
+
+export const SaasAccountProvider = ({ children }: { children: ReactNode }) => {
+  if (!import.meta.env.SAAS_URL) {
+    return <>{children}</>
+  }
+  const saasAcount = useSaasAccountProvider()
```

Comment:
Two things here:
- returning `children` (with no fragment nor brackets) should be enough
- hooks should not be instanced after a conditional rendering, it should be called before. Since it is a hook from a react-query call, you should be passing a boolean to this specific one using the `enable` feature.